docs/modules/ROOT/pages/tmail-backend/jmap-extensions/pushWithFirebase.adoc

```diff
@@ -0,0 +1,210 @@
+= Push with Firebase
+:navtitle: Push with Firebase
+
+Allows devices to register PUSH though link:https://firebase.google.com/[FireBase Cloud Messaging], and manage its
+registrations. The server will then push `StateChanges` through FireBase to the client.
+
+== Additions to the capability object
+
+Servers supporting the Push with firebase extension need
+to advertize it through the session capabilities:
+
+....
+com:linagora:params:jmap:firebase:push
+....
+
+The associated object is empty.
+
+== Encoding of state change into firebase payload
+
+Once registered, the JMAP server will send StateChange objects to firebase using the token of the Firebase registration.
+
+See https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#Message.FIELDS.data for Firebase message format.
+
+StateChange is encoded into the *data* part of the payload.
+
+Each property is composed of the *accountId* followed by a delimiter `:` and then the `Type`. Associated to this property
+is the `state` of that type for the given account. Implementors need to be careful to split on the last delimitor as
+accountId might contain the delimiter.
+
+=== Example
+
+The following state change:
+
+....
+{
+  "@type": "StateChange",
+  "changed": {
+    "a3123": {
+      "Email": "d35ecb040aab",
+      "EmailDelivery": "428d565f2440",
+      "CalendarEvent": "87accfac587a"
+    },
+    "a43461d": {
+      "Mailbox": "0af7a512ce70",
+      "CalendarEvent": "7a4297cecd76"
+    }
+  }
+}
+....
+
+will be encoded as:
+
+....
+{
+  "data": {
+    "a3123:Email": "d35ecb040aab",
+    "a3123:EmailDelivery": "428d565f2440",
+    "a3123:CalendarEvent": "87accfac587a"
+    "a43461d:Mailbox": "0af7a512ce70",
+    "a43461d:CalendarEvent": "7a4297cecd76"
+  },
+  "token": "fcm-client-token"
+}
+....
+
+== The FirebaseRegistration object
+
+A FirebaseRegistration object has the following fields:
+
+- **id**: `Id`, server set, immutable, the JMAP identifier of the object.
+- **token**: (immutable) Set by the client, it allows firebase to route notifications to him.
```

Comment:
back-end should handle the case when this token is expired https://firebase.google.com/docs/cloud-messaging/manage-tokens

Reply:
That's done through expiracy property described below.